packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -225,6 +234,10 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
         }
         // TODO t15106389: add helper utility for fetching more data
         this._pendingRefetch = null;
+        this._relayContext = {
+          environment: this.context.relay.environment,
+          variables: fragmentVariables
```

Comment:
will this be a bit too expensive ? as in [`cwr`](https://github.com/facebook/relay/blob/master/packages/react-relay/modern/ReactRelayRefetchContainer.js#L114)  we will restart everything over again once context variable changes;

I may be wrong, but I think  [think line](https://github.com/facebook/relay/blob/master/packages/react-relay/modern/ReactRelayRefetchContainer.js#L128)  might be a better place to plug in the fix. 
my understanding is that : A child's `props` is part of its parents `state`
process is like:
1. parent call refetch
2. parent receive new data and call `setstate` with updated data
3. child gets new props 
4. child should be re-rendered 

but if child has done its own refetch , then *by definittion* it has the latest of everything that's why we had a `!this._localVariables` check. as @josephsavona  said before